### PR TITLE
feat(color): Display current stick position on CurveEdit screen

### LINKED ISSUE #2508: Stick position indicator on the curve graph setting.
### Is there an existing issue for this feature request?

- [X] I have searched the existing issues

### Is your feature request related to a problem?

There is no stick position indicator on the chart when setting the curves function.

### Describe the solution you'd like

Is it possible to get a pointer to the stick position on a curve graph setting, similar to the graph in the input menu?
It is very useful to see the position of the stick on the chart when adjusting the curves function.

### Describe alternatives you've considered

_No response_

### Additional context

RM  TX16S  2.8.0-RC2

### PR DESCRIPTION
**Pull request by eshifri**
Fixes #2508 

This PR adds functionality to display stick position on the edit curve screen.
Functionality is similar to OTX and ETX B/W: if you enter CurveEdit from input or mixer the stick position is shown; if you go to this screen from CURVES tab - no cursor is shown on the curve.
(It is possible to keep "last used stick", but it does not look right to me.)

### CODE REVIEW

**Comment by raphaelcoeffic** [comment]
> It is possible to keep "last used stick", but it does not look right to me

You mean using auto-source? The solution implemented here looks quite cumbersome to me. It assumes the source of the last edited item should be used. What if I have not edited anything?

**Comment by eshifri** [comment]
As description says, this works only if you enter the EditCurve from edit input or edit mixer.
If the EditCurve is entered from CURVES, the currentInput is set to zero and no cursor is shown.
What I have meant by  "It is possible to keep last used stick", is we can remove this set to zero and use the sticjk from the previous edit. If such is not available - still no cursor. 
This does not look as a right behavior to me, thus there is no cursor if we enter the curve edit from the CURVES.
This is the same behavior we currently have for B/W radios and OTX implemented for color lcd radios in 2.4.14.

**Comment by Eldenroot** [comment]
I believe this is the first step just to get the functionality for color radios as well. The limitation now will be same for bw and color radios. 

Anyway this should be extended and cover all places in GUI. Otherwise it could be confusing... I am not asking in this PRs, but in future... if @eshifri can work on this one :)

Thank you!

**Comment by pfeerick** [comment]
> The solution implemented here looks quite cumbersome to me. It assumes the source of the last edited item should be used. What if I have not edited anything?

This is at is a valid assumption to make, since the premise is the same as current B&W behaviour - if you access the curves screen via input or mixes, then the configured source for the curve input should be used. However, if you enter via the curves screen (or via an input with no configured source (?!), then yes, there is no source, and thus the next level - an option that lets you pick a source, is valid.

**Comment by raphaelcoeffic** [comment]
@pfeerick fine with me, then I'll leave it in your skilled hands and probably make a further PR then to add auto-source.

**Comment by eshifri** [comment]
This commit adds automatic source selection.
Now the source is locked if the CurveEdit is entered from inputs or mixers and uses auto source selection if curve editor is entered from CURVES.

**Comment by pfeerick** [comment]
Seems to be working perfectly on TX16S - uses configured source if curve editor accessed via inputs or mixes, otherwise uses whichever source is moved. 

I have found an edge case where it stops working - if you go into the curve editor via inputs or mixes, graph moves when you move the stick. If you then go into the curve editor directly via the curves tab, autosource doesn't work. But it does if you go in the other order. 

In other words, accessing the curves via inputs/mixes breaks autosource on the curve editor tab.

**Comment by pfeerick** [comment]
This seemed to have fixed it... I think it was still locked?
```diff
diff --git a/radio/src/gui/colorlcd/curveedit.cpp b/radio/src/gui/colorlcd/curveedit.cpp
index a6afc9115..29418c841 100644
--- a/radio/src/gui/colorlcd/curveedit.cpp
+++ b/radio/src/gui/colorlcd/curveedit.cpp
@@ -127,6 +127,8 @@ void CurveEdit::SetCurrentSource(uint32_t source)
   CurveEdit::currentSource = source;
   if (source) 
     lockSource = true;
+  else
+    lockSource = false;
 }
 
 mixsrc_t CurveEdit::currentSource = 0;
```

**Comment by eshifri** [comment]
Yes, I think this is the proper fix.

**Comment by eshifri** [comment]
Many thanks to @pfeerick for finding the problem and the solution!

**Comment by pfeerick** [comment]
Great! And you're welcome! :)

**Comment by Alex6892** [comment]
> Большой! Добро пожаловать! :)

Thanks for the work done, How can I use it, there is a ready-made nightly firmware for RM TX16S?

**Comment by pfeerick** [comment]
Once a PR is merged, it's in the next nightly build ;) https://github.com/EdgeTX/edgetx/releases/tag/nightly

**Comment by Alex6892** [comment]
> Как только PR объединен, он находится в следующей ночной сборке;) https://github.com/EdgeTX/edgetx/releases/tag/nightly

Thank you very much, I will test.

**Comment by Alex6892** [comment]
> > Как только PR объединен, он находится в следующей ночной сборке;) https://github.com/EdgeTX/edgetx/releases/tag/nightly
> 
> Thank you very much, I will test.

Hello.
I found a cursor movement issue in the Curve menu.
It occurs on an ascending-descending curve and only if the curve is used in the "inputs" menu.
Please watch the video, I tried to point out the problem in detail.
Thank you for your attention and your work.

https://disk.yandex.ru/i/6Pnn4Vq4n3D7Bw

**Comment by eshifri** [comment]
I can reproduce it on the simulator. 
And it happens for both Curves and Mixes.
The underlying problem is the source jumps momentarily from 4 to 87 for Rudder and from 1 to 90 for Ailerons.
And only if the curve is used by the corresponding input.

If the curve is used by Rudder source detector occasionally returns 87 instead of 4 when the stick is near zero crossing. 
For Ailerons it jumps from 1 to 90.

Source detector problem?
